src/Log/AccessLogFactory.php

```diff
@@ -38,12 +39,21 @@ public function __invoke(ContainerInterface $container) : AccessLogInterface
         $config = $config['zend-expressive-swoole']['swoole-http-server']['logger'] ?? [];
 
         return new Psr3AccessLogDecorator(
-            $container->has(LoggerInterface::class) ? $container->get(LoggerInterface::class) : new StdoutLogger(),
+            $this->getLogger($container, $config),
             $this->getFormatter($container, $config),
             $config['use-hostname-lookups'] ?? false
         );
     }
 
+    private function getLogger(ContainerInterface $container, array $config) : LoggerInterface
+    {
+        if (isset($config['logger-name'])) {
+            return $container->get($config['logger-name']);
```

Comment:
I have considered making further checks here (that the container `has` the service and such), but I have finally decided to just provide whatever is found so that an explicit exception is thrown either if the service is not found or if it does not return a `LoggerInterface` instance.

That will make debugging much easier, in case of an error, like a typo in the service name.